ddtrace/contrib/asgi/middleware.py

```diff
@@ -128,21 +116,31 @@ async def __call__(self, scope, receive, send):
         if sample_rate is not None:
             span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, sample_rate)
 
+        method = scope.get("method")
+        server = scope.get("server")
+        url = None
```

Comment:
```suggestion
```

IMO it's easier to reason about code when the variable is explicitly defined under each conditional branch:

```python
if some_condition:
    url = something.url
else:
    url = None
```
this shows a clear intention when defining the variable as opposed to

```python
url = None

if some_condition:
    url = something.url
```